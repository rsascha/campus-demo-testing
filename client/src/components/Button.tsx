type DataType = {
  data: string;
};

type ButtonProps = {
  text: string;
  color?: string;
  onClick?: (data: DataType) => void;
};

export function Button({ text, onClick }: ButtonProps) {
  function handleOnClick() {
    if (onClick) {
      onClick({ data: "some data" });
    }
  }

  return (
    <button onClick={handleOnClick} data-testid="my-button-component">
      {text}
    </button>
  );
}
