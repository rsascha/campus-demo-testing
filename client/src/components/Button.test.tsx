import { render } from "@testing-library/react";
import { beforeAll, describe, expect, it, vi } from "vitest";
import { Button } from "./Button";

describe("Button", () => {
  let onClick: () => void;
  let button: HTMLElement;

  beforeAll(() => {
    onClick = vi.fn();
    const renderResult = render(<Button text="click me" onClick={onClick} />);
    button = renderResult.getByTestId("my-button-component");
  });

  it("should render a button", () => {
    expect(button).toBeDefined();
  });

  it("should render a button and match snapshot", () => {
    expect(button).toMatchSnapshot();
  });

  it("should call onClick when button is clicked", () => {
    button.click();
    expect(onClick).toHaveBeenCalledWith({ data: "some data" });
  });
});
