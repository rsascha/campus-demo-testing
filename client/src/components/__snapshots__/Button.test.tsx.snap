// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render a button and match snapshot 1`] = `
<button
  data-testid="my-button-component"
>
  click me
</button>
`;
